saleor/graphql/order/utils.py

```diff
@@ -27,7 +31,35 @@ def validate_total_quantity(order: "Order", errors: T_ERRORS):
         )
 
 
-def validate_shipping_method(order: "Order", errors: T_ERRORS):
+def get_shipping_method_availability_error(
+    order: "Order",
+    method: Optional["ShippingMethodData"],
+    manager: "PluginsManager",
+):
+    """Validate whether shipping method is still available for the order."""
+    valid_methods_ids = (
+        {
+            method.id
+            for method in get_valid_shipping_methods_for_order(
```

Comment:
Can we revert this part of code a little bit? 
There is no need to call `get_valid_shipping_methods_for_order` when `method` is None, right? Can we check this as a first thing?